pkg/events/event.go.go

```diff
@@ -0,0 +1,114 @@
+package events
+
+import (
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type Event struct {
+	id           ID // TODO: use id ?
+	id32Bit      ID
+	name         string
+	docPath      string // Relative to the 'doc/events' directory
+	internal     bool
+	syscall      bool
+	dependencies Dependencies
+	sets         []string
+	params       []trace.ArgMeta
+}
+
+func NewEvent(name string, sets []string, depsID []ID) Event {
+	evt := Event{
+		id32Bit: Sys32Undefined,
+		name:    name,
+		sets:    sets,
+	}
+
+	d := Dependencies{
+		Events: make([]ID, 0, len(depsID)),
+	}
+
+	d.Events = append(d.Events, depsID...)
+
+	evt.dependencies = d
+
+	return evt
+}
+
+// NewEvent creates a new Event with all its fields set.
+func NewEventFull(
```

Comment:
maybe we can have only this constructor, named `NewEvent` and the other `NewEvent`which seems to be used to create an event from signature is forced to pass all values?

Reply:
Yes this was in my original PR and it is definitely something we need (I'll make that change). I was actually making that change yesterday before my last push.